linchpin/provision/roles/libvirt/tasks/copy_image_src_remote.yml

```diff
@@ -48,41 +48,41 @@
 - name: "set image_src for uncompressed files"
   set_fact:
     img_src: "{{ local_image_src }}"
-  when: lismt_remote.stat.mime_type == "application/octet-stream"
+  when: lismt_remote.stat.mimetype == "application/octet-stream"
```

Comment:
I'm assuming there's a good reason for this change? The mime_type still works as an alias according to [the ansible stat module (v2.5)](http://docs.ansible.com/ansible/latest/modules/stat_module.html). Even older versions still have mime_type listed under get_mime. Though I did see a bug reported that the documentation was incorrect, but not yet fixed.

Reply:
As seen below the output of the stat module have no mime_type keyword:

    "msg": {
        "changed": false,                                                                                                                                                                                          
        "failed": false,                                                                  
        "stat": {                                                                                                                                                                                                  
            "atime": 1527073187.537445,                                                                                                                                                                            
            "attr_flags": "e",                                                                                                                                                                                     
            "attributes": [                                                                                                                                                                                        
                "extents"                                                                                                                                                                                          
            ],                                                                                                                                                                                                     
            "block_size": 4096,                                                                                                                                                                                    
            "blocks": 5857544,                                                                                                                                                                                     
            "charset": "binary",                                                                                                                                                                                   
            "checksum": "09bb9738b2903e9eefd64f69e3836d052fc292ec",                                                                                                                                                
            "ctime": 1527073198.1374464,                                        
            "dev": 64770,                                                                                                                                                                                          
            "device_type": 0,                                                                                                                                                                                      
            "executable": false,                                                               
            "exists": true,                                                                                                                                                                                        
            "gid": 1001,                                                                                                                                                                                           
            "gr_name": "admin",                                                                                                                                                                                    
            "inode": 11,                                                                                                                                                                                           
            "isblk": false,                                                                                                                                                                                                    "ischr": false,                                                                                                                                                                                        
            "isdir": false,                                                                                                                                                                                        
            "isfifo": false,                                                                                                                                                                                       
            "isgid": false,                                                                                                                                                                                        
            "islnk": false,                                                                                                                                                                                        
            "isreg": true,                                                                                                                                                                                         
            "issock": false,                                                                                                                                                                                       
            "isuid": false,                                                                                                                                                                                        
            "mimetype": "application/octet-stream",                                                                                                                                                                
            "mode": "0664",                                                       
            "mtime": 1527073187.535445,                                                                                                                                                                            
            "nlink": 1,                                                                                                                                                                                            
            "path": "/var/lib/libvirt/images/linchpin//consul-2018-05-17.qcow2",                                                                                                                                   
            "pw_name": "admin",                                                                                                                                                                                    
            "readable": true,                                                                                                                                                                                      
            "rgrp": true,                                                                                                                                                                                          
            "roth": true,                                                                                                                                                                                          
            "rusr": true,                                                                                                                                                                                          
            "size": 2999058432,                                                                                                                                                                                   
            "uid": 1001,                                                                                                                                                                                           
            "version": "18446744073194595580",                                                                                                                                                                     
            "wgrp": true,                                                                                                                                                    
            "woth": false,                                                              
            "writeable": true,                                                                                                                                                                                     
            "wusr": true,                                                                                                                                                                                          
            "xgrp": false,                                                                                                                                                                                         
            "xoth": false,                                                                                                                                                                                         
            "xusr": false                                                                                                                                                                                          
        }                                                                               
    }                              

Changing the conditional expression back to

when: lismt_remote.stat.mime_type == "application/octet-stream"'           

results in the following error:

    fatal: [localhost]: FAILED! => {"msg": "The conditional check 'lismt_remote.stat.mime_type == \"application/octet-stream\"' failed. The error was: error while evaluating conditional (lismt_remote.stat.mime_type 
== \"application/octet-stream\"): 'dict object' has no attribute 'mime_type'\n\nThe error appears to have been in '/home/joanbe/github/linchpin/linchpin/provision/roles/libvirt/tasks/copy_image_src_remote.yml': 
line 51, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: \"set image_src for uncompressed files\"\n  ^ here\n"}

Reply:
What version of ansible are you using here? It might behoove us to ensure back to our minimum version (I believe ansible 2.3.1 at the moment) that this aligns. I'll add a task to write a basic test here to test on different ansible versions.

Reply:
Running ansible inside the virtualenv gives:
$ ansible --version
ansible 2.5.2
  config file = /home/joanbe/.virtualenvs/linchpin-src/ansible.cfg
  configured module search path = [u'/home/joanbe/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/joanbe/.virtualenvs/linchpin-src/lib/python2.7/site-packages/ansible
  executable location = /home/joanbe/.virtualenvs/linchpin-src/bin/ansible
  python version = 2.7.15 (default, May 16 2018, 17:50:09) [GCC 8.1.1 20180502 (Red Hat 8.1.1-1)]

Reply:
Having mimetypes like this should work for the moment since it passes all of the tests.